rag_experiment_accelerator/evaluation/plain_metrics.py

```diff
@@ -1,153 +1,163 @@
+        fuzzy_match_fn = getattr(fuzz, match_type)
+    except AttributeError:
+        raise ValueError(f"Match type '{match_type}' is not recognized.")
+
+    similarity_score = fuzzy_match_fn(str1, str2)
     return similarity_score
 
 
-def levenshtein(value1, value2):
+def levenshtein(str1: str, str2: str) -> int:
     """
     Calculates the Levenshtein distance between two strings and returns the normalized similarity score as a percentage.
 
     Args:
-        value1 (str): The first string to compare.
-        value2 (str): The second string to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
 
     Returns:
         int: The normalized similarity score as a percentage.
     """
-    score = int(algorithms.levenshtein.normalized_similarity(value1, value2) * 100)
+    score = d.Levenshtein.normalized_similarity(str1, str2) * 100
     return score
 
 
-def jaccard(value1, value2):
+def jaccard(str1: str, str2: str) -> int:
     """
     Calculates the Jaccard similarity score between two sets of values.
 
     Args:
-        value1 (set): The first set of values.
-        value2 (set): The second set of values.
+        str1 (set): The first set of values.
+        str2 (set): The second set of values.
 
     Returns:
         int: The Jaccard similarity score between the two sets of values, as a percentage.
     """
-    score = int(algorithms.jaccard.normalized_similarity(value1, value2) * 100)
+    score = int(alg.jaccard.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def hamming(value1, value2):
+def hamming(str1: str, str2: str) -> int:
     """
     Calculates the Hamming similarity score between two values.
 
     Args:
-        value1 (str): The first value to compare.
-        value2 (str): The second value to compare.
+        str1 (str): The first value to compare.
+        str2 (str): The second value to compare.
 
     Returns:
         int: The Hamming similarity score between the two values, as a percentage.
     """
-    score = int(algorithms.hamming.normalized_similarity(value1, value2) * 100)
+    score = int(d.Hamming.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def jaro_winkler(value1, value2):
+def jaro_winkler(str1: str, str2: str) -> int:
     """
     Calculates the Jaro-Winkler similarity score between two strings.
 
     Args:
-        value1 (str): The first string to compare.
-        value2 (str): The second string to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
 
     Returns:
         int: The Jaro-Winkler similarity score between the two strings, as an integer between 0 and 100.
     """
-    score = int(algorithms.jaro_winkler.normalized_similarity(value1, value2) * 100)
+    score = int(d.JaroWinkler.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def cosine(value1, value2):
+def cosine(str1: str, str2: str) -> int:
     """
```

Comment:
Severity: Major - The `cosine` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.